api/src/main/java/org/apache/iceberg/view/ViewRepresentation.java

```diff
@@ -19,14 +19,21 @@
 package org.apache.iceberg.view;
 
 import java.util.Locale;
+import org.immutables.value.Value;
 
+@Value.Immutable
 public interface ViewRepresentation {
 
   enum Type {
```

Comment:
can we potentially avoid the enum type and directly use string? That seems to be able to simplify things a bit. (this should probably be asked in the API PR, but I guess it's still not too late).

Reply:
Agreed, see the discussion here on this topic of enums in the original API PR https://github.com/apache/iceberg/pull/4925/files#r1015167642

I think as long as we have well defined and documented type strings that should be acceptable.